app/decorators/resource_decorator.rb

```diff
@@ -32,6 +32,12 @@ def display_doi
     MintingStatusDoiComponent.new(resource: resource_with_doi)
   end
 
+  def display_rights
+    return unless respond_to?(:rights)
+
+    DisplayRightsComponent.new(id: rights)
+  end
```

Comment:
@srerickson I'm curious what errors you were seeing when this was on the WorkVersionDecorator. My guess is that if this is defined on the WorkVersionDecorator, you won't need the return statement, but, I bet you'll need to account for nil ids. That may be why you were seeing errors. See my other comment about the `render?` method. That may solve the problem.

Reply:
@awead here's an example of the error that happens if I move it to WorkVersionDecorator. (This is with the render? method added).

```rb
  1) WorkVersionMetadataComponent rendering renders a string with label
     Failure/Error: value = format_value(work_version.send(attr))
     
     NoMethodError:
       undefined method `display_rights' for #<ResourceDecorator:0x0000562f16f02800>
     # ./app/components/work_version_metadata_component.rb:58:in `block in attributes'
     # ./app/components/work_version_metadata_component.rb:56:in `map'
     # ./app/components/work_version_metadata_component.rb:56:in `attributes'
     # ./app/components/work_version_metadata_component.html.erb:3:in `call'
     # /usr/local/bundle/gems/view_component-2.18.0/lib/view_component/base.rb:78:in `render_in'
     # /usr/local/bundle/gems/view_component-2.18.0/lib/view_component/render_component_helper.rb:6:in `render_component'
     # /usr/local/bundle/gems/view_component-2.18.0/lib/view_component/test_helpers.rb:27:in `render_inline'
     # ./spec/components/work_version_metadata_component_spec.rb:7:in `block (2 levels) in <top (required)>'
     # ./spec/components/work_version_metadata_component_spec.rb:35:in `block (3 levels) in <top (required)>'
```